presto-spi/src/main/java/io/prestosql/spi/connector/ConnectorMetadata.java

```diff
@@ -419,6 +419,11 @@ default ConnectorInsertTableHandle beginInsert(ConnectorSession session, Connect
         throw new PrestoException(NOT_SUPPORTED, "This connector does not support inserts");
     }
 
+    default ConnectorInsertTableHandle beginInsert(ConnectorSession session, ConnectorTableHandle tableHandle, List<ColumnHandle> columnsToBeInserted)
```

Comment:
are we going to support both ways? Which one is prefered? Should we deprecate (or even remove) the other `beginInsert` method?

Reply:
We can deprecate the other `beginInsert` method and then remove it

Reply:
Why not to remove it now?

Reply:
It might be a backward incompatible change so was planning to remove them after we moved all the connector API to that model. Can we handle it in the same PR or ?

Reply:
No. Let's just deprecate the other method.